### PR TITLE
separate utils

### PR DESCRIPTION
**Pull request by fehiepsi**
@neerajprad I'll add docs/comments in the last PR.

### CODE REVIEW

File: test/test_hmc_util.py [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+
+@pytest.mark.parametrize("num_steps, expected", [
+    (18, [(0, 17)]),
+    (50, [(0, 6), (7, 44), (45, 49)]),
+    (100, [(0, 14), (15, 89), (90, 99)]),
+    (150, [(0, 74), (75, 99), (100, 149)]),
+    (200, [(0, 74), (75, 99), (100, 149), (150, 199)]),
+    (280, [(0, 74), (75, 99), (100, 229), (230, 279)]),
+])
+def test_build_adaptation_schedule(num_steps, expected):
```
Comment (neerajprad) [inline test/test_hmc_util.py]: I thought we already had these tests? Should we remove the older tests.

Comment (fehiepsi) [reply test/test_hmc_util.py]: @neerajprad I just rename the files so no new tests are added.

Comment (fehiepsi) [reply test/test_hmc_util.py]: Oppss. Wait, I need to separate the tests from distribution utils too.

Comment (fehiepsi) [reply test/test_hmc_util.py]: done!

**Pull request merged**